pallets/order-book/src/lib.rs

```diff
@@ -248,20 +287,49 @@ pub mod pallet {
         #[pallet::weight(<T as Config>::WeightInfo::create_orderbook())]
         pub fn create_orderbook(
             origin: OriginFor<T>,
-            _dex_id: T::DEXId,
+            dex_id: T::DEXId,
             order_book_id: OrderBookId<T>,
-            _tick_size: OrderPrice<T>,
-            _step_lot_size: OrderVolume<T>,
-            _min_lot_size: OrderVolume<T>,
-            _max_lot_size: OrderVolume<T>,
         ) -> DispatchResult {
-            ensure_root(origin)?;
+            let who = ensure_signed(origin)?;
+            ensure!(
+                order_book_id.base_asset_id != order_book_id.target_asset_id,
+                Error::<T>::ForbiddenToCreateOrderBookWithSameAssets
+            );
+            let dex_info = T::DexInfoProvider::get_dex_info(&dex_id)?;
+            ensure!(
+                order_book_id.base_asset_id == dex_info.base_asset_id,
+                Error::<T>::NotAllowedBaseAsset
+            );
+            T::AssetInfoProvider::ensure_asset_exists(&order_book_id.target_asset_id)?;
+            T::EnsureTradingPairExists::ensure_trading_pair_exists(
+                &dex_id,
+                &order_book_id.base_asset_id.into(),
+                &order_book_id.target_asset_id.into(),
+            )?;
             ensure!(
                 !<OrderBooks<T>>::contains_key(order_book_id),
                 Error::<T>::OrderBookAlreadyExists
             );
-            // todo (m.tagirov)
-            todo!()
+
+            let order_book =
+                if T::AssetInfoProvider::get_asset_info(&order_book_id.target_asset_id).2 != 0 {
+                    // regular asset
+                    OrderBook::<T>::default(order_book_id, dex_id)
```

Comment:
Maybe it's better to use something `new_regular` and `new_nft` method names? Or even add builder for OrderBook struct.

For example
```rust
OrderBookBuilder::default().nft().with_id(order_book_id).with_dex_id(dex_id)
```

Reply:
I don't like an idea to implement Default trait for OrderBook, because I don't want to allow create some default order books without `order_book_id` and `dex_id`.
I think `default_nft` is better than `new_nft` because it better represents that we have default settings inside.
If you want I can rename `default` to `default_regular` 😇

Reply:
> If you want I can rename default to default_regular not to be confused with Default trait

Yeah, I think it's enough